app.py

```diff
@@ -92,289 +82,212 @@
-        with connection_manager as manager:
-            with manager.get_cursor() as cursor:
-                # Sample userId from the database
-                username = auth.username()
+        async with connection_manager as manager:
+            async with manager.get_cursor() as cursor:
+                username = credentials.username
                 logger.info(f"Fetching user ID for username: {username}")
-                db_user = asyncio.run(get_user(cursor, username))
+                db_user = await get_user(cursor, username)
                 user_id = db_user.id
 
                 container_client = ContainerClient.from_connection_string(
                     FERTISCAN_STORAGE_URL, container_name=f"user-{user_id}"
                 )
 
-                # Get JSON form from the request
-                form = request.get_json()
-
                 if form is None:
-                    logger.warning("Missing form in request")
-                    return jsonify(
-                        error="Missing fertiliser form!"
-                    ), HTTPStatus.BAD_REQUEST
-
-                files = request.files.getlist("images")
+                    raise HTTPException(status_code=HTTPStatus.BAD_REQUEST, detail="Missing fertiliser form!")
 
                 # Collect files in a list
-                images = []
-                for file in files:
-                    if file:
-                        logger.info(f"Reading file: {file.filename}")
-                        images.append(file.stream.read())
+                images = [await file.read() for file in files if file]
 
                 logger.info(f"Registering analysis for user_id: {user_id}")
-                inspection = asyncio.run(
-                    fertiscan.register_analysis(
-                        cursor=cursor,
-                        container_client=container_client,
-                        user_id=user_id,
-                        analysis_dict=form,
-                        hashed_pictures=images,
-                    )
+                inspection = await fertiscan.register_analysis(
+                    cursor=cursor,
+                    container_client=container_client,
+                    user_id=user_id,
+                    analysis_dict=form,
+                    hashed_pictures=images,
                 )
 
-                logger.info(
-                    f"Inspection created successfully with id: {inspection['inspection_id']}"
-                )
-                return jsonify(inspection), HTTPStatus.CREATED
+                logger.info(f"Inspection created successfully with id: {inspection['inspection_id']}")
+                return inspection
 
     except Exception as err:
         logger.error(f"Error occurred: {err}")
         logger.error("Traceback: " + traceback.format_exc())
-        return jsonify(error=str(err)), HTTPStatus.INTERNAL_SERVER_ERROR
+        raise HTTPException(status_code=HTTPStatus.INTERNAL_SERVER_ERROR, detail=str(err))
```

Comment:
Write a test function for this route using the TestClient of FastAPI.

The test should check if an error if returned if there's no inspection in the request.